dirty_cat/_fuzzy_join.py

```diff
@@ -218,43 +242,93 @@ def fuzzy_join(
         main_col = right_col
         aux_col = left_col
 
-    # Make sure that the column types are string and categorical:
-    main_col_clean = main_table[main_col].astype(str)
-    aux_col_clean = aux_table[aux_col].astype(str)
+    # Check if all included columns are numeric:
+    if (
+        main_table.select_dtypes(include="number")
+        .columns.isin(list(main_table[main_col].columns))
+        .any()
+    ):
+        is_numeric = True
+    else:
+        is_numeric = False
+
+    if len(main_col) == 1 and len(aux_col) == 1 and is_numeric is False:
+        main_col = main_col[0]
+        aux_col = aux_col[0]
+
+    if metric in ["number"] and is_numeric:
+        neigh = NearestNeighbors(n_neighbors=1)
+        aux_array = aux_table[aux_col].to_numpy()
+        main_array = main_table[main_col].to_numpy()
+        # Reweighting to avoid measure specificity
+        aux_array = minmax_scale(aux_array)
+        main_array = minmax_scale(main_array)
```

Comment:
We need a 2-norm (standard scaler), I believe, and not a minmax scaling.

Reply:
Thanks, definitely works better.